devtools/cli/src/main/java/io/quarkus/cli/plugin/CliPluginsAdd.java

```diff
@@ -0,0 +1,109 @@
+    Optional<String> description;
+
+    @CommandLine.Parameters(arity = "1", paramLabel = "PLUGIN_NAME", description = " The plugin location (e.g. url, path or maven coordinates in GACTV form)")
+    String location;
+
+    PluginCatalogService pluginCatalogService = new PluginCatalogService();
+
+    @Override
+    public Integer call() {
+        try {
+            output.debug("Add plugin with initial parameters: %s", this);
+            output.throwIfUnmatchedArguments(spec.commandLine());
+
+            if (runMode.isDryRun()) {
+                dryRunAdd(spec.commandLine().getHelp());
+                return CommandLine.ExitCode.OK;
+            }
+
+            return addPlugin();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return output.handleCommandException(e,
+                    "Unable to add plugin(s): " + name + " of type: " + type.map(PluginType::name).orElse("<any>") + "."
+                            + e.getMessage());
+        }
+    }
+
+    Integer addPlugin() throws IOException {
+        File catalogFile;
+        PluginCatalog catalog;
+        reconcile(output);
+        String name = this.name.orElse(PluginUtil.getName(location));
+
+        Optional<PluginType> type = this.type
+                .or(() -> Optional.ofNullable(getAvailablePlugins().get(name)).map(Plugin::getType))
+                .or(() -> Optional.of(location).map(PluginUtil::getType));
+
+        Optional<Path> matchingBinary = type
+                .flatMap(t -> (t == PluginType.binary) ? Binaries.pathOfComamnd(name).map(File::toPath) : Optional.empty());
+        Optional<String> location = matchingBinary.map(Path::toAbsolutePath).map(Path::toString)
+                .or(() -> Optional.of(this.location));
+
+        if (catalogOptions.project) {
+            catalogFile = pluginCatalogService.findProjectCatalogPath(Optional.ofNullable(output.getTestDirectory()))
+                    .map(Path::toFile)
+                    .orElseThrow(() -> new IllegalStateException("Could not determine project catalog path."));
+        } else {
+            catalogFile = pluginCatalogService.getUserCatalogPath(catalogOptions.userDirectory).toFile();
+        }
+
+        if (!catalogFile.exists() && !catalogFile.getParentFile().mkdirs() && !catalogFile.createNewFile()) {
+            throw new IOException("Failed to create project catalog at: " + catalogFile.getAbsolutePath());
+        }
+
+        catalog = pluginCatalogService.readCatalog(catalogFile.toPath());
+        Plugin plugin = new Plugin(name, type.orElse(PluginType.jbang), location, description,
+                isAutoremovable(name, type));
+        catalog = catalog.addPlugin(plugin);
+        pluginCatalogService.writeCatalog(catalog);
+        return CommandLine.ExitCode.OK;
+    }
+
+    void dryRunAdd(CommandLine.Help help) {
+        output.printText(new String[] {
```

Comment:
should this not be printed even in case of non dryrun?

Reply:
Agree, we need more feeback.